public_html/wp-content/themes/pattern-directory/src/components/pattern/index.js

```diff
@@ -1,17 +1,28 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
 /**
  * Internal dependencies
  */
 import PatternPreview from '../pattern-preview';
 import PatternPreviewActions from '../pattern-preview-actions';
 import ReportPatternButton from '../report-pattern-button';
+import { store as patternStore } from '../../store';
 
-const Pattern = ( { content, postId, userHasReported, loggedIn } ) => {
+const Pattern = ( { postId, userHasReported, loggedIn } ) => {
 	// postId as passed from the HTML dataset is a string.
 	postId = Number( postId ) || 0;
+	const pattern = useSelect( ( select ) => select( patternStore ).getPattern( postId ), [ postId ] );
+	if ( ! pattern ) {
+		return null;
+	}
+
 	return (
 		<>
 			<PatternPreviewActions postId={ postId } />
-			<PatternPreview blockContent={ content } />
+			<PatternPreview blockContent={ pattern.content.rendered } />
```

Comment:
Is it possible that `rendered` is not present?

Reply:
No, if the pattern exists, then it was a successful API request which has `content.rendered`.